### ISSUE TITLE
Copy-to with key links to original rather than to copy

### ISSUE THREAD

**Issue by robander** [dita-ot/dita-ot#2842]
## Expected Behavior
If I have these key definitions:

```
<topicref href="alsocopykey.dita" keys="also1"/>
<topicref href="alsocopykey.dita" copy-to="alsocopykey-newname.dita" keys="also2"/>
```

Key references to `also1` should resolve to `alsocopykey.dita`, while key references to `also2` should resolve to the copied version, `alsocopykey-newname.dita`.

## Actual Behavior

Key references to both `also1` and `also2` resolve to the original topic, `alsocopykey.dita`. When I generate HTML5, two topics are generated but all links are to the original; when I generate PDF, the topic appears twice but all links are to the original.

I also have a reference that appears *only* with copy-to, and is never referenced without the copy-to:
`<topicref href="onlycopykey.dita" copy-to="onlycopykey-newname.dita" keys="only"/>`

For that case, the HTML5 build generates both `onlycopykey.html` and `onlycopykey-newname.html`, but all uses of `keyref="only"` resolve to the original name. In the PDF, `<xref keyref="only"/>` results in blue text but no link, because the original file `onlycopykey.dita` is not in the PDF to provide a link destination.

## Possible Solution

When the `keyref` value is resolved for key definitions that set `@copy-to`, the link should go to the copy specified in `@copy-to`. At this point don't yet know if that change is simple or would require a giant architectural change.

## Steps to Reproduce

Sample map includes both variants described above. Tested with DITA-OT 3.0 and 2.5.

[copykey.zip](https://github.com/dita-ot/dita-ot/files/1459173/copykey.zip)

This is related to #2841 (and is probably the preferred way to get the same result). Not sure if this overlaps #2826 at all, as my sample doesn't use branch filtering but does have a key that resolves to copy-to value.

**Comment by raducoravu** [comment]
👍 One of our end users also complained about the same thing.

**Comment by robander** [comment]
Chatting further with @jelovirt about this as I tried to think through a solution. It felt like resolving `@copy-to` very early and updating `@href` to the new value -- before resolving keys -- would fix the problem, because at that point you resolve to the right target. But resolving `@copy-to` before keys is not possible if you also allow this markup:
`<topicref keyref="a" copy-to="b.dita"/>`

**Comment by robander** [comment]
Fixed with #2935
